include/swift/AST/DiagnosticsSema.def

```diff
@@ -1601,6 +1601,10 @@ ERROR(inferred_opaque_type,none,
 // Extensions
 ERROR(non_nominal_extension,none,
       "non-nominal type %0 cannot be extended", (Type))
+WARNING(composition_in_extended_type,none,
+        "extending a protocol composition is not supported; extending %0 "
+        "instead. Use a 'where' clause for additional constraints.",
```

Comment:
We try to avoid multiple sentences in diagnostics (see docs/Diagnostics.md). Do we really need the `where` clause bit?